src/StateSubject.js

```diff
@@ -0,0 +1,61 @@
+import { Subject } from 'rxjs/Subject';
+
+// Used as a placeholder value so we can tell
+// whether or not the real state has been set,
+// even if the real state was `undefined`
+const UNSET_VALUE = {};
+
+// We can't use BehaviorSubject because when we create
+// and give state$ to your epics your reducers have not
+// yet run, so there is no initial state yet until the
+// `@@redux/INIT` action is emitted. BehaviorSubject
+// requires an initial value (which would be undefined)
+// and if an epic subscribes to it on app boot it would
+// synchronously emit that undefined value incorrectly.
+export class StateSubject extends Subject {
+  constructor(store) {
+    super();
+    // If you're reading this, keep in mind that this is
+    // NOT part of the public API and will be removed!
+    this.__store = store;
+    this._value = UNSET_VALUE;
+  }
+
+  _subscribe(subscriber) {
+    const subscription = super._subscribe(subscriber);
+    if (this._value !== UNSET_VALUE && subscription && !subscription.closed) {
+      subscriber.next(this._value);
+    }
+    return subscription;
+  }
+
+  get value() {
+    if (this._value === UNSET_VALUE) {
+      if (process.env.NODE_ENV !== 'production') {
+        require('./utils/console').warn('You accessed state$.value inside one of your Epics, before your reducers have run for the first time, so there is no state yet. You\'ll need to wait until after the first action (@@redux/INIT) is dispatched or by using state$ as an Observable.');
```

Comment:
Wordsmithing on this warning is welcome. It's a hard thing to describe. I'm referring to this:

```js
const somethingEpic = (action$, state$) => {
  state$.value // <------ oops
  // etc...
};
```

But less obviously it often happens like this:

```js
const somethingEpic = (action$, state$) => {
  return action$.ofType(PING)
    .mapTo({
      type: PONG,
      value:   state$.value // <------ oops
    });
};
```

Reply:
I think linking to the FAQ is a great idea!

Any way we could use template literals here so I can review it on multiple lines? 😉

Reply:
I've read the message over a couple times and it makes perfect sense to me.

Reply:
I suppose I'd write this in less of a "you" style:

> One of the epics used `state$.value` before the reducers have run for the first time, so there is no state yet. The `state$.value` can't be used until after the first action (`@@redux/INIT`) is dispatched or by using `state$` as an Observable.